KOGITO-2777 Add trustyAI input-data and model-lookup sections

kelvah opened the pull request:
Link to [Jira ticket](https://issues.redhat.com/browse/KOGITO-2777).

This PR adds input data and model lookup sections of the execution detail page.

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

kelvah commented on ui-packages/yarn.lock:
```diff
@@ -2748,6 +2806,11 @@
   dependencies:
     "@types/node" ">= 8"
 
+"@patternfly/patternfly@2.71.5":
+  version "2.71.5"
+  resolved "https://registry.yarnpkg.com/@patternfly/patternfly/-/patternfly-2.71.5.tgz#db096c07cf3478a3c05f9020e3cd16be9c098be2"
+  integrity sha512-kkPVr+B9UKDeko8F31RzO7g1ecMbp4rVVQhg6X95rrzPd9mqBC8TGuwhyXnnHCbSAyNyqsp+ldGhDMsN5yMPiw==
+
```
This really old PatternFly version is required by `@kogito-tooling/embedded-editor`. Its usage is confined to the iframe of the embedded editor.

manstis replied:
We can probably move `kogito-apps/ui-packages/packages/trusty` to `@kogito-tooling/embedded-editor@0.6.1` 

This uses `@patternfly/patternfly@4.16.7` see [here](https://github.com/kiegroup/kogito-tooling/blob/master/yarn.lock#L2002).

I'll try it locally first.

manstis reviewed (approved):
LGTM, but trusty's `prepare` script did not run following a `yarn install` (even with `node_modules` deleted). I had to run it manually to copy the GWT editor and _envelope_ to the `static` folder.

manstis commented:
@kelvah FYI, I changed the dependency for `@kogito-tooling/embedded-editor` to `@0.6.0` and everything still works OK.

IDK if you want to move to the newer version of `kogito-tooling` (IMO it'd be advisable).

jstastny-cz commented on ui-packages/packages/trusty/src/components/Templates/InputData/useInputData.tsx:
```diff
@@ -0,0 +1,35 @@
+    httpClient(config)
+      .then(response => {
+        if (isMounted) {
+          setInputData({ status: 'SUCCESS', data: response.data.inputs });
```
the status field is not from axios here? cause there it's defined as number ...

kelvah replied:
Almost. Here the status field is a different one used to save the request status inside `setInputData`. The instruction is inside the `then` callback, which is executed when the request succeeds.

jstastny-cz replied:
Ah, ok, resolving the comment then.

jstastny-cz reviewed (commented):
I've noticed a few generated files being committed here. Is that expected? I'd expect we rely on the generation.

kelvah commented:
> @kelvah FYI, I changed the dependency for `@kogito-tooling/embedded-editor` to `@0.6.0` and everything still works OK.
> 
> IDK if you want to move to the newer version of `kogito-tooling` (IMO it'd be advisable).

I've updated `@kogito-tooling/embedded-editor` to `0.6.0`. Thank you.

I've also added a `prepare` script to the root `package.json` because yarn workspaces is not running npm lifecycle scripts at package level. Now the embedded editor should be automatically extracted after installation.

kelvah commented on ui-packages/packages/trusty/package.json:
```diff
@@ -27,9 +29,13 @@
     "test:coverage": "yarn test --coverage",
     "test:watch": "jest --watch",
     "lint": "tslint -c ./tslint.json --project . './src/**/*.ts{,x}'",
+    "prepare": "yarn run copy-embedded-editor",
```
The `prepare` script is one of the predefined [npm life-cycle scripts](https://docs.npmjs.com/misc/scripts). It gets executed automatically after the install script. Which is the right moment to extract the embedded editor.

jstastny-cz replied:
Ah, so this will run on each build of the ui-packages maven module, right?
That might lead to excessive time needed for build of unrelated projects (e.g. management-console), when they're being run separately. I am not aware of the details here, regarding the size of downloaded artifact or how the changes affect the build speed - please provide your view.

The current setup is the following:
* ui-packages - maven project to build ALL the UI packages using yarn
* quarkus-apps (e.g. kogito-apps/management-console) that take the bundle from ui-packages/packages/management-console/dist and creating a executable jar.
* the quarkus app has maven dependency on `ui-packages` to build the react apps first when building just a single module instead of whole repo - that's when the build time takes the hit.

If we decide this change is affecting our build time in local dev workflow (not sure if anyone actually works with the project that way) we can deal with this in a separate enhancement.

kelvah replied:
Yes, it will run on each build of ui-packages.
I don't see any performance issue though. The `prepare` script is way faster then the install one for example.
It takes ~2s on my local env.

jstastny-cz replied:
that might be due to the fact it is cached (maven command using -nsu).

Just checked the PR check - in clean environment the kie-bc-editors-unpacked maven build takes  17.100 s , that seems reasonable. As I mentioned, it can be tweaked later.

Thanks for the explanation.

manstis commented:
Jenkins retest this

manstis commented on ui-packages/packages/trusty/pom.xml:
```diff
@@ -0,0 +1,159 @@
+    <artifactId>kie-bc-editors-unpacked</artifactId>
+    <groupId>org.kie.kogito</groupId>
+    <packaging>jar</packaging>
+    <version>0.0.0</version>
```
This copies the approach (100%) in `kogito-tooling`. See [here](https://github.com/kiegroup/kogito-tooling/blob/master/packages/kie-bc-editors-unpacked/pom.xml). If there are recommendations to be made that can be applied across all of this type of use-case I think a JIRA reporting the issue and proposing a solution would be great.

manstis commented on ui-packages/packages/trusty/pom.xml:
```diff
@@ -0,0 +1,159 @@
+    </repositories>
+
+    <properties>
+        <version.dmn.webapp>7.39.0-SNAPSHOT</version.dmn.webapp>
```
Correct.. and it's probably not a bad thing; as trusty can tie itself to a stable editor release if needed.

`kogito-tooling` takes an identical approach (updating the version as needed). See [here](https://github.com/kiegroup/kogito-tooling/commits/master/packages/kie-bc-editors-unpacked/pom.xml).

Admittedly we could probably move to `7.42.0-SNAPSHOT` in this PR.

jstastny-cz replied:
Thanks for clarifying, I'll leave to you the decision which version to have here.

danielezonca reviewed (approved):
LGTM 👍

manstis commented:
Hi @mbiarnes @elguardian @cristianonicolai @r00ta IDK if the failure of this [build](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/organizations/jenkins/KIE%2Fkogito%2Fghprb-webhooks%2Fkogito-apps/detail/kogito-apps/1160/pipeline) is of interest to you `com.google.cloud.tools.jib.http.ResponseException: 429 Too Many Requests. "Too Many Requests. Please see https://docs.docker.com/docker-hub/download-rate-limit/"`. I suspect either the job or a test is downloading repeatedly from Docker Hub (either for this repository in isolation or across `kie` repositories?)

manstis commented:
Jenkins retest this

kelvah commented:
Jenkins retest this

kelvah commented:
@manstis I think we may have an issue with `cxsd`. The last build failed at `yarn install` because:
```
[INFO] make: g++: Command not found
[INFO] make: *** [Release/obj.target/node_expat/node-expat.o] Error 127
```

`cxsd` requires `node-expat`, which relies on `node-gyp` for compilation. So [it needs gcc](https://github.com/nodejs/node-gyp#installation) to work and jenkins seems not to have it installed.

How should we proceed? Is the _temporary_ removal of `cxsd` a possibility? Its generated files are under vcs.

manstis commented:
Hi @kelvah, please feel free to remove all reference to `cxsd` and keep just the generated sources.

It might be useful to put a `README` pointing to [here](https://github.com/manstis/pmml-codegen/commit/b2d8f8a231044c9c4ec19cbb6084787cc04c9d84) like I have for `jsonix` in the PMML marshaller [too](https://github.com/kiegroup/kogito-tooling/blob/master/packages/pmml-editor-marshaller/src/marshaller/model/README.md).

kelvah commented:
hi @manstis, I'm afraid it's not enough. [`unmarshaller.ts`](https://github.com/kiegroup/kogito-apps/pull/417/files#diff-17a5a65c6c758015412b346308aac26e) is using `cxml`, a transitive dep of `cxsd`. and it has the same compilation issue.

manstis commented:
@kelvah oh dear :-(

The use of `cxsd` is replaced by the new PMML marshaller in `kogito-tooling` (see my open PR that is first in line for merge but needs QE approval). Therefore I suggest killing everything relating to `cxsd` in your PR meaning we'll have no Linear Regression Viewer for now (please leave files somewhere I can get them again later when my PR is merged and I can complete the changes).

It's a public holiday in tbe UK today but I'll be checking emails from time to time.

kelvah commented:
Jenkins retest this

jstastny-cz reviewed (approved):
LGTM
